_posts/2018-12-29-dask-on-hpc.md

```diff
@@ -0,0 +1,139 @@
+{% include JB/setup %}
+
+We analyze large datasets on HPC systems with Dask, a library for scalability
+that works well with the existing Python software ecosystem,
+and works comfortably with native HPC hardware.
+
+This article explains why this makes sense for us.
+Our motivation is to share our experiences with our colleagues,
+and to highlight opportunities for future work.
+
+We'll start with seven reasons why we use Dask,
+followed by seven common issues that affect us today.
+
+
+## Reasons why we use Dask
+
+### 1.  Ease of use
+
+Dask extends libraries like Numpy, Pandas, and Scikit-learn, which are well
+known APIs for scientists and engineers.  It also extends simpler APIs for
+multi-node multiprocessing.  This makes it easy for our existing user base to
+get up to speed.
+
+By abstracting the parallelism away from the user/developer, our analysis tools
+can be written by computer science non-experts, such as the scientists
+themselves, meaning that our software engineers can take on a more supporting
+role than a leadership role.
+
+
+### 2.  Smooth HPC integration
+
+With tools like [Dask Jobqueue](https://jobqueue.dask.org) and [Dask MPI](https://mpi.dask.org)
+there is no need of any boilerplate bash code with `qsub/sbatch/...`.
+
+Dask interacts natively with our existing job schedulers (SLURM/SGE/SGE/LSF/...)
+so there is no additional system to set up and manage between users and IT.
+All the infrastructure that we need is already in place.
+
+Also, interactive analysis at scale and auto scaling is just awesome, and lets
+us use our existing infrastructure in new ways and improves our occupancy
+generally.
+
+
+### 3.  Aimed for scientific processing
+
+Dask is compatible with scientific data formats like HDF5, NetCDF, Parquet, and
+so on.  This is because Dask works with other libraries within the Python
+ecosystem, like XArray, which already has strong support for scientific data
+formats and processing.
+
+
+### 4.  Versatility of APIs
+
+And yet Dask is not designed for any particular workflow, and instead can
+provide infrastructure to cover a variety of different problems within an
+institution.  Everything is possible.
+
+
+### 5.  Versatility of Infrastructure
+
+Dask is compatible with laptops, servers, HPC systems, and even cloud
+computing if needed.  The environment can change with very little code
+adaptation.
+
+Dask is more than just a tool to us; it is a gateway to thinking about a
+completely different way of providing computing infrastructure to our users.
+Dask opens up the door to cloud computing technologies (such as elastic scaling
+and object storage) and makes us rethink what an HPC center should really look
+like!
+
+### 6.  Cost and Collaboration
+
+Dask is free and open source, which means we do not have to rebalance our
+budget and staff to address the new immediate need of data analysis tools.
+There are no licenses, and we have the ability to make changes when necessary.
+
+
+# What needs work
+
```

Comment:
At the moment, guidelines for adequate strategies or tutorials on prototypical calculations are limited to my opinion and could be one item in this list.

I understand the dashboard is here to help you decide on you own how to adapt your computations. From experience, this is not as simple as it seems and multiple trial and errors is the norm in order to perform *large scale* calculations (rechunking on O(10TB) data is typical computation in my group for example).

One issue that we have to face typically on HPC platforms is that the dataset does not or does barely fit in memory.
My experience tells me this has profound implications on the strategies you have to develop compared to platforms where the dataset easily fits in memory.